### ISSUE TITLE
Panel mode + auto-entities + Map: card is too high

### ISSUE THREAD

**Issue by ildar170975** [thomasloven/lovelace-auto-entities#436]
2024.4.1
1.12.1

Consider this card:
```
type: custom:auto-entities
card:
  type: map
filter:
  include:
    - entity_id: device_tracker.virtual_tracker_1
```
on a Panel view.

The card is too high - scrollbars appear, the map in not properly centered:
Initially you even cannot see the device_tracker's marker:
![image](https://github.com/thomasloven/lovelace-auto-entities/assets/71872483/892acd5e-098c-4378-bbc2-265fe855f51e)

After zooming:
![image](https://github.com/thomasloven/lovelace-auto-entities/assets/71872483/ff306cf3-4896-4f32-8115-9dc48471fff5)

Note a scrollbar - the card now is not "limited by a viewport".

Compare with a Map card w/o auto-entities:
![image](https://github.com/thomasloven/lovelace-auto-entities/assets/71872483/94a1645a-64e8-4fec-bb7b-4d480fa34449)

Compare with 2023.7.1 + 1.12.1 (Map + auto-entities) - same issue, i.e. it takes place in an old HA version as well:
![image](https://github.com/thomasloven/lovelace-auto-entities/assets/71872483/b65e4576-f21f-485a-9d33-6cd76bca4c72)

**Comment by Mariusthvdb** [comment]
as the +/- buttons in the screenshot above remain visible, I didnt notice this issue before.

But, this morning in beta, I suddenly noticed that when clicking these buttons only once, they move out of the view:

before:

<img width="690" alt="Scherm­afbeelding 2024-06-27 om 11 10 10" src="https://github.com/thomasloven/lovelace-auto-entities/assets/33354141/041c1843-cb2f-45d9-b469-4f05e3fab572">

after clicking either + or -"

<img width="682" alt="Scherm­afbeelding 2024-06-27 om 11 10 22" src="https://github.com/thomasloven/lovelace-auto-entities/assets/33354141/1d2c9862-1a8f-4cc6-a0ce-d5d16efda81c">

and the only way to get them back is by reloading the view. the center button does not help.

config:

```
title: Kaart
path: kaart
icon: mdi:map
type: panel
cards:

  - type: custom:auto-entities
    filter:
      include:
        - domain: person
    card:
      type: map
      hours_to_show: 48
```

listing my person entities verbosely fixes the issue 100%

**Comment by lingfish** [comment]
Yep, I have this issue too, have for months.

**Comment by Cyberes** [comment]
yup, same here

**Comment by Cyberes** [comment]
Same here. @thomasloven @ildar170975

**Comment by lingfish** [comment]
I just thought I'd ping you @thomasloven on this old issue -- any thoughts?

**Comment by Grumblezz** [comment]
I'm able to reproduce the issue as well  with HA 2025.8.3 and Auto entities version 1.16.1. 

My use case is where the amount of zones is expanded quite frequently based on a certain POI (for instance with supermarkets). So being able to track whether someone is at a certain zone 🛒 (PIO) is really useful and used on a daily basis. And as I often lost the ability to click on these buttons in the top left corner, I found that manually adding the entities fixes it, but it is a nightmare 👻👻 to maintain. My configuration is: 

```
views:
  - theme: Backend-selected
    title: Locations
    icon: mdi:map
    type: panel
    cards:
      - type: custom:auto-entities
        filter:
          include:
            - options: {}
              domain: person
            - options: {}
              domain: zone
          exclude:
            - options: {}
              hidden_by: user
        card:
          type: map
          hours_to_show: 24
        card_param: entities
    path: map
```

**Comment by lingfish** [comment]
I've come back to this due to frustration. It would seem it also happens without auto at all; the blame lies with `type: vertical-stack` in my case... if that's removed, issue gone but then the map doesn't fill the available screen size :(

**Comment by Mariusthvdb** [comment]
might be a silly suggestion, but, given you wont have hundreds of person entities, why not just list those and be done with it?

has been super reliable since I did so [1.5 years ago](https://github.com/thomasloven/lovelace-auto-entities/issues/436#issuecomment-2194204940).

sometimes it wont work as it is supposed to, then you got to use it the way it does.....

**Comment by lingfish** [comment]
> might be a silly suggestion, but, given you wont have hundreds of person entities, why not just list those and be done with it?

You're not considering other entities... like zones. Sure, might be a few people entities, but a lot more zones. Not to mention when new ones are added, having them auto appear in the map(s) rather than having to remember to update it is appealing too.

> sometimes it wont work as it is supposed to, then you got to use it the way it does.....

Yeah, or... report bugs.  As I said in my most recent comment, it might not be `lovelace-auto-entites` fault though.

**Comment by dcapslock** [comment]
> I've come back to this due to frustration. It would seem it also happens without auto at all; the blame lies with `type: vertical-stack` in my case... if that's removed, issue gone but then the map doesn't fill the available screen size :(

@lingfish can you share a minimal config? I checked the issue thread and can only see a 'me too' without any config. So if you need your specific case reviewed, which may be different as you mention another card, it would be best to share a config.

**Comment by lingfish** [comment]
> can you share a minimal config? I checked the issue thread and can only see a 'me too' without any config. So if you need your specific case reviewed, which may be different as you mention another card, it would be best to share a config.

Yeah @dcapslock, that's fair. Here we go, some stuff obfuscated of course:

```yaml
title: Map
views:
  - theme: Backend-selected
    title: X only
    path: map_X_only
    cards:
      - type: vertical-stack
        cards:
          - type: custom:mushroom-entity-card
            title: X ETA
            entity: input_boolean.update_X_travel_time
          - type: conditional
            conditions:
              - condition: state
                entity: input_boolean.update_X_travel_time
                state: 'on'
            card:
              type: gauge
              name: X ETA
              max: 60
              needle: true
              entity: sensor.google_travel_time
          - type: custom:auto-entities
            card:
              type: map
              hours_to_show: 24
              theme_mode: auto
            entities:
              - entity: person.x
            filter:
              include:
                - domain: zone
            show_empty: false
    visible:
      - user: ***
    type: panel
```

The idea is that this shows a toggle to turn on updating `google_travel_time`, and a conditional to show a nice gauge of travel time ETA... and then a map.

**Comment by dcapslock** [comment]
> Yeah [@dcapslock](https://github.com/dcapslock), that's fair. Here we go, some stuff obfuscated of course:

Thanks. Can you confirm what is the issue? I see that when once I scroll to the map that it then takes control and you can't scroll back up. Is this what you are trying to solve?

EDIT: On Chrome desktop if you hover over scrollbar area you can grab the scrollbar and scroll. On mobile if the width of the panel is less safe areas so you always have background to grab to scroll.

I also copied your config and took out AE. Just the same.

**Comment by dcapslock** [comment]
I am finding no difference between Map only and AE+Map. So while I see that there is issues with using a map in a panel, I can't see that it's an AE issue. 

Issues I see:
1. When Zooming, map scrolls to top without any reference to header, so controls get covered.
2. Once you Zoom on the map, you can't pan to scroll the page. Looks to be the map is using greedy gestures rather than cooperative. (See [Googles take](https://developers.google.com/maps/documentation/javascript/interaction) on this). HA is using leaflet but there is likely same approach.

**Comment by lingfish** [comment]
EDIT: added Android screenshot.

So the key issue is that the map, in my opinion, should fill the "viewport" (is what I'd call it), and not present vertical scrollbars, as if the map "div" is a fixed height. It should fit to the remaining window real estate.

https://github.com/user-attachments/assets/409d4559-3798-435e-abdb-5d1a0335d2de

It does this for example in the default map view.

Additionally, on mobile (Android at least), it renders too short, vertically.

![Image](https://github.com/user-attachments/assets/6f5c9bf0-7e55-453e-b84d-796e33e6a78c)

> I can't see that it's an AE issue.

Yeah, as I said earlier, it seems to be a `vertical-stack` issue, not AE.

**Comment by dcapslock** [comment]
I am not quite following. Is the video what is meant to be, or the issue?

**Comment by lingfish** [comment]
> I am not quite following. Is the video what is meant to be, or the issue?

It's the issue. On desktop, it should fill the current window vertically and not have scrollbars.

On mobile, it should also fill, and not stop short of available screen size.

**Comment by dcapslock** [comment]
Got it. Thanks.

**Comment by dcapslock** [comment]
OK. My take is whether it is a map in a stack card or a custom card, the HA map resize observer will resize the map to try (?) and be square, with the width being the driving dimension. i.e. if width is smaller than client height, don't make bigger, but if width is greater than client height, make the map square with it being as high as wide. Same basis as why it does not likely stretch on mobile in that again the resize oberserver is not quite right. But when the map is in a pabel by itself in a panel, it will resize to the panel. So its a Frontend issue, not AE. Suggest you take it to Frontend as an issue, presenting your testing with a vertical stack (they will close if you use custom card as an example). In reality, there are probably a lot of things to get right with your wish to have both a stack and a full viewport map. Being able to have coopertaive gestures etc. being another map aspect to get right.

**Comment by Mariusthvdb** [comment]
I just did another test of [my failing config from last year](https://github.com/thomasloven/lovelace-auto-entities/issues/436#issuecomment-2194204940)
which ws also used by @Grumblezz [ in his issue](https://github.com/thomasloven/lovelace-auto-entities/issues/436#issuecomment-3250790688),

```
title: Kaart
path: kaart
icon: mdi:map
type: panel
cards:
  - type: custom:auto-entities
    card:
      type: map
      hours_to_show: 24
    entities:
    filter:
      include:
        - domain: zone
        - domain: person
```

but that is running perfectly currently.
So that particular config should probably also not be taken into consideration anymore
